network/network_linux.go

```diff
@@ -408,10 +408,13 @@ func (nm *networkManager) connectExternalInterface(extIf *externalInterface, nwI
 	*/
 	isGreaterOrEqualUbuntu17 := isGreaterOrEqaulUbuntuVersion(ubuntuVersion17)
 	if isGreaterOrEqualUbuntu17 {
-		log.Printf("[net] Saving dns config from %v", extIf.Name)
-		if err = saveDnsConfig(extIf); err != nil {
-			log.Printf("[net] Failed to save dns config: %v", err)
-			return err
+		// Don't copy dns servers if systemd-resolved isn't available
+		if _, cmderr := platform.ExecuteCommand("systemctl status systemd-resolved"); cmderr == nil {
```

Comment:
instead of shelling out to systemctl, should we use [dbus](https://github.com/coreos/go-systemd/blob/main/dbus/methods.go#L469)?

Reply:
what happens if systemd-resolved is available but not running? should we do systemctl check systemd-resolved and verify if its active?

Reply:
dbus would give us that granularity of information :)

Reply:
[this](http://refspecs.linuxfoundation.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic.html#INISCRPTACT) should be good enough in our case (checking whether systemd-resolved is active or not)